os_net_config/impl_ifcfg.py

```diff
@@ -1984,6 +1989,12 @@ def apply(self, cleanup=False, activate=True, config_rules_dns=True):
                         )
                         self.ifdown(interface_name)
                         self.remove_config(ifcfg_file)
+            for nmconn_file in glob.iglob(nm_cleanup_pattern()):
+                interface_name = Path(nmconn_file).stem
+                cmd = ['nmcli', 'connection', 'delete', interface_name]
```

Comment:
Can you also look for any dispatcher scripts present in `/etc/NetworkManager/dispatcher.d/<scripts>` and remove the same. There are some scripts created by nmstate in this path. May be this could be handled in a separate PR.

Reply:
I've not seen anything else creating issues with deployment.